src/components/S3Viewer/FileListView.tsx

```diff
@@ -0,0 +1,116 @@
+import { FC, MouseEvent, useEffect, useMemo, useState } from 'react';
+import { S3Object } from '../../types/S3Object';
+import { Table, TableBody, TableContainer, TablePagination } from '@mui/material';
+import EnhancedTableHead from '../EnhancedTableHead';
+import { ObjectRow } from './ObjectRow';
+import { S3Client } from '@aws-sdk/client-s3';
+
+interface FileListViewProps {
+  client: S3Client;
+  bucket: string;
+  objects: S3Object[];
+  permissions: any;
```

Comment:
would we be able to specify a type for the permissions?

Reply:
Sure I can do that